### PR TITLE
Use Random Number instead of Timestamp for Nonce

### PR DESCRIPTION
**Pull request by danintel**
Timestamps have a greater chance of collision, as use of
parallel transaction processing increases (either through
threads or multiple processes).

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

File: sdk/examples/noop_python/sawtooth_noop/client_cli/create_batch.py [hunk 1]
```diff
@@ -31,7 +30,7 @@
 class NoopPayload(object):
     def __init__(self):
         self.nonce = binascii.b2a_hex(random.getrandbits(
-            8 * 8).to_bytes(8, hbyteorder='little'))
+            8 * 8).to_bytes(8, byteorder='little'))
```
Comment (vaporos) [inline sdk/examples/noop_python/sawtooth_noop/client_cli/create_batch.py]: This is a different type of change than the rest of the commit; suggest it be in a different commit with an alternate explanation if it's not an accident.

Comment (danintel) [reply sdk/examples/noop_python/sawtooth_noop/client_cli/create_batch.py]: As requested, I reintroduced the hbyteorder typo.

Then I separated the s/hbyteorder/byteorder/ fix into a separate fix.

**Comment by danintel** [comment]
Rebased as this PR is waiting in the queue.
